x/stakers/keeper/msg_server_claim_commission_rewards.go

```diff
@@ -22,17 +21,18 @@ func (k msgServer) ClaimCommissionRewards(goCtx context.Context, msg *types.MsgC
 	}
 
 	// Check if amount can be claimed
-	if staker.CommissionRewards < msg.Amount {
-		return nil, types.ErrNotEnoughRewards.Wrapf("%d > %d", msg.Amount, staker.CommissionRewards)
+	if !msg.Amount.IsAllLTE(staker.CommissionRewards) {
```

Comment:
low prio: msg.Amount.IsAnyGT(staker.CommissionRewards) is probably more readable

Reply:
Also wanted to use this before but using `msg.Amount.IsAnyGT(staker.CommissionRewards)` is actually dangerous here because isAnyGT returns false if msg.Amount has a coin which is not in staker.CommissionRewards. This would then pass and panic in `staker.CommissionRewards = staker.CommissionRewards.Sub(msg.Amount...)`, therefore IsAllLTE has to be used here since it checks if all denoms are also present.

Description of IsAnyGT:

> IsAnyGT returns true iff for any denom in coins, the denom is present at a greater amount in coinsB.
e.g. {2A, 3B}.IsAnyGT{A} = true {2A, 3B}.IsAnyGT{5C} = false {}.IsAnyGT{5C} = false {2A, 3B}.IsAnyGT{} = false